lib/datadog/grok/src/parse_grok.rs

```diff
@@ -786,4 +791,37 @@ mod tests {
             ),
         ]);
     }
+
+    #[test]
+    fn parses_with_new_lines() {
+        test_full_grok(vec![
+            (
+                "%{data:field}",
+                "a\nb",
+                Ok(Value::from(btreemap! {
+                    "field" => "a\nb"
+                })),
+            ),
+            (
+                "%{data:line1}\n%{data:line2}",
+                "a\nb",
+                Ok(Value::from(btreemap! {
+                    "line1" => "a",
+                    "line2" => "b"
+                })),
+            ),
+            // disable DOTALL mode
+            ("(?-m)%{data:field}", "a\nb", Err(Error::NoMatch)),
```

Comment:
Do we want that users are able to change the regex modes, is that an explicit feature?

Reply:
Yes, they can do it as with "normal" regular expressions.